lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -1325,6 +1281,69 @@ void PatchEntryPointMutate::addUserDataArg(SmallVectorImpl<UserDataArg> &userDat
   userDataArgs.push_back(UserDataArg(argTy, name, userDataValue, argIndex));
 }
 
+// =====================================================================================================================
+// Add a PushConst UserDataArg
+//
+// @param userDataArgs : Vector to add UserDataArg to
+// @param node : PushConst resource node
+// @param builder : IRBuilder (just for getting types)
+void PatchEntryPointMutate::addPushConstArgs(SmallVectorImpl<UserDataArg> &userDataArgs, const ResourceNode &node,
+                                             llvm::IRBuilder<> &builder) {
```

Comment:
It looks a little suspicious to me that you can just use the same code here regardless of whether the node is in the root table or a descriptor table.  I expect that you will need to different function that will share some similar code.

Reply:
> To make sure that the implementation actually works, you could also add an amber testcase like the one for a16 (https://github.com/GPUOpen-Drivers/llpc/blob/dev/test/amber/a16.amber). It may be a bit hidden in the a16 example, but they support lit CHECK lines to check for lines in the pipeline dumps.
> 
> (These tests are not yet run in CI unfortunately, but you can run them locally with `ninja check-amber` in an xgl build.)

Thanks, I try to add it.